identity-account-storage/src/storage/traits.rs

```diff
@@ -118,6 +120,31 @@ pub trait Storage: storage_sub_trait::StorageSendSyncMaybe + Debug {
   /// Returns `true` if a key exists at the specified `location`.
   async fn key_exists(&self, did: &IotaDID, location: &KeyLocation) -> Result<bool>;
 
+  /// Encrypts the given `data` with the specified `algorithm`
+  ///
+  /// Diffie-Helman key exchange will be performed in case an [`KeyType::X25519`] is given.
+  async fn encrypt_data(
```

Comment:
Please expand on the documentation here. Give some examples (in words) what the parameters `did` and `associated_data` might be used for.